### PR TITLE
Fixes/recurring aos fetch retry

### PR DESCRIPTION
**Pull request by tekwani**

### CODE REVIEW

File: lib/ws_servers/api/send_recurring_algo_order_list.js [hunk 1]
```diff
@@ -6,17 +6,36 @@ const sendError = require('../../util/ws/send_error')
...
+  let algoOrders = []
+  let retries = 0
+  let receivedData = false
+  let err = null
+  while (!receivedData && retries < 3) {
+    try {
+      algoOrders = await restPaginationHandler(rest.getRecurringAlgoOrders.bind(rest), {})
+      receivedData = true
+      err = null
+      debug('fetched total Recurring AOs', algoOrders.length)
```
Comment (tarcisiozf) [inline lib/ws_servers/api/send_recurring_algo_order_list.js:19]: return `algoOrders` after debug so we can avoid a whole logic check.

File: lib/ws_servers/api/send_recurring_algo_order_list.js [hunk 2]
```diff
@@ -6,17 +6,36 @@ const sendError = require('../../util/ws/send_error')
...
  *
  * @param {RESTv2} rest
  * @returns {Promise} p
  */
-const getAndUpdateRecurringAlgoOrdersList = async (rest, algoDB) => {
-  const algoOrders = await restPaginationHandler(
-    rest.getRecurringAlgoOrders.bind(rest),
-    {}
-  )
+const getAndUpdateRecurringAlgoOrdersList = async (rest, algoDB, debug) => {
```
Comment (tarcisiozf) [inline lib/ws_servers/api/send_recurring_algo_order_list.js:37]: new args not included in the jsdoc above

**Review by tarcisiozf — CHANGES_REQUESTED** [review]

File: lib/ws_servers/api/send_recurring_algo_order_list.js (hunk 1 above)
Comment (tekwani) [reply lib/ws_servers/api/send_recurring_algo_order_list.js:16]: Fixed.

File: lib/ws_servers/api/send_recurring_algo_order_list.js (hunk 2 above)
Comment (tekwani) [reply lib/ws_servers/api/send_recurring_algo_order_list.js:37]: Fixed

**Review by tarcisiozf — APPROVED** [review]

**Pull request merged**